ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_15/ExplicitDisclosureIT.scala

```diff
@@ -477,7 +477,7 @@ final class ExplicitDisclosureIT extends LedgerTestSuite {
 
   test(
     "EDInconsistentSuperfluousDisclosedContracts",
-    "The ledger accepts superfluous disclosed contracts with mismatching payload",
+    "The ledger rejects superfluous disclosed contracts with mismatching payload",
```

Comment:
Did the behavior change? If yes, can you also update the test to assert that? (I see that the test is not expecting any error 🤔 )

Reply:
Good catch. 
There is indeed a change of behavior, but my original change did not really capture the change.
Basically all the disclosed contract are check but the engine as much as it can. In particular it check the contracts type check. So I add to change the the two first cases to be properly type and check the last one which is not valid anymore.

I add additional check in `EDMalformedDisclosedContracts` to capture the new check the engine is doing.